apps/dashboard/src/components/header-navigation/edit-bridge-url-button.tsx

```diff
@@ -0,0 +1,123 @@
+import { useLayoutEffect, useState } from 'react';
+import { RiLinkM, RiPencilFill } from 'react-icons/ri';
+import { PopoverPortal } from '@radix-ui/react-popover';
+import { SubmitHandler, useForm } from 'react-hook-form';
+// eslint-disable-next-line
+// @ts-ignore
+import { zodResolver } from '@hookform/resolvers/zod';
+import * as z from 'zod';
+
+import { cn } from '@/utils/ui';
+import { Popover, PopoverContent, PopoverTrigger } from '../primitives/popover';
+import { Button } from '../primitives/button';
+import { Input, InputField, InputGroup, InputHint, InputLabel } from '../primitives/input';
+import { useBridgeHealthCheck, useUpdateBridgeUrl, useValidateBridgeUrl } from '@/hooks';
+import { ConnectionStatus } from '@/utils/types';
+import { useEnvironment } from '@/context/environment/hooks';
+
+type FormFields = {
+  bridgeUrl: string;
+};
+
+const schema = z.object({ bridgeUrl: z.string().url() });
+
+export const EditBridgeUrlButton = () => {
```

Comment:
This is the component that allows editing the Bridge URL, it includes:
- popover shown on click
- the pulsing indicator for the bridge health check with the corresponding logic
- form to update URL with validation logic